core/services/kraken/manifest/manifest.py

```diff
@@ -0,0 +1,134 @@
+import dataclasses
+import datetime
+from typing import List, Optional
+
+# Extra
+import aiohttp
+from dataclass_wizard import fromdict
+
+# Models
+from manifest.models import ExtensionVersion, RepositoryEntry
+
+# Current Extension Manifest host
+REPO_URL = "https://bluerobotics.github.io/BlueOS-Extensions-Repository/manifest.json"
+
+
+@dataclasses.dataclass
+class ManifestContent:
+    """
+    Represents the manifest content with cache control.
+
+    Attributes:
+        extensions (List[RepositoryEntry]): List of extensions in the manifest.
+        issued_at (Optional[int]): Timestamp when the manifest was issued.
+        expires_in (Optional[int]): Time in seconds until the manifest expires.
+    """
+
+    extensions: List[RepositoryEntry]
+    expires_in: int = 3600
+    issued_at: int = int(datetime.datetime.now(datetime.timezone.utc).timestamp())
+
+    @property
+    def is_expired(self) -> bool:
+        """
+        Checks if the manifest is expired.
+
+        Returns:
+            bool: True if the manifest is expired, False otherwise.
+        """
+
+        return (int(datetime.datetime.now(datetime.timezone.utc).timestamp()) - self.issued_at) > self.expires_in
+
+
+class Manifest:
+    """
+    Class responsible for fetching and managing the extension manifest.
+    """
+
+    _instance: Optional["Manifest"] = None
+    _cached_manifest: Optional[ManifestContent] = None
+
+    def __init__(self):
+        raise RuntimeError("This class should not be instantiated, use Manifest.instance() instead")
+
+    @classmethod
+    def instance(cls) -> "Manifest":
+        """
+        Returns the instance of the manifest manager.
+
+        Returns:
+            Manifest: Instance of the manifest.
+        """
+
+        if cls._instance is None:
+            cls._instance = cls.__new__(cls)
+        return cls._instance
+
+    async def _fetch_manifest(self) -> ManifestContent:
+        """
+        Fetches the manifest from the repository.
+
+        Returns:
+            ManifestContent: Manifest content.
+
+        Raises:
+            RuntimeError: If the manifest could not be fetched.
+        """
+
+        async with aiohttp.ClientSession() as session:
+            async with session.get(REPO_URL) as resp:
```

Comment:
I know it was already implemented, but maybe it's a good idea to add a `, headers={'Accept': 'application/json'}`?

Reply:
Added!